lib/goodies/conduit/plug/dead_letter.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Goodies.Conduit.Plug.DeadLetter do
+  use Conduit.Plug.Builder
+
+  @moduledoc """
+  Publishes messages that were nacked or raised an exception to a
+  dead letter destination and ack them to be removed from the regular queue.
+
+  Based on original `Conduit.Plug.DeadLetter`.
+
+  The following options are required:
+
+    * `broker` - The broker to publish the message with.
+    * `publish_to` - The place to publish the message.
+
+  In order for this to work, you must also define an outgoing publish with the same
+  name as the `publish_to` option in your broker.
+
+      publish :error, to: "my_app.error"
+
+  ## Examples
+
+      plug Goodies.Conduit.Plug.DeadLetter, broker: MyApp.Broker, publish_to: :error
+
+  """
+
+  def init(opts) do
+    # Fail if opts are missing
+    _ = Keyword.fetch!(opts, :publish_to)
+    _ = Keyword.fetch!(opts, :broker)
```

Comment:
```elixir
_ = Keyword.fetch(opts, :publish_to) or raises "Missing required option publish_to"
```
o que de fazer algo assim? deixaria o erro mais explicito

Reply:
@guimunarolo eu não alterei a implementação atual pra não fugir mto do que já existe conduit caso eles atualizem o código ou algo nesse sentido. mas seguindo essa ideia, o que temos hj é algo similar a
```elixir
** (KeyError) key :publish_to not found in: [test: "some value"]
    (elixir 1.10.4) lib/keyword.ex:399: Keyword.fetch!/2
```
e vc está propondo (`Keyword.fetch/2`) algo que seria similar a
```elixir
** (RuntimeError) Missing required option publish_to
```
se a gente complementar essa mensagem do raise faz sentido, talvez algo do tipo `DeadLetter plug missing required option :publish_to, received #{inspect(opts)}`

EDIT: chamar `opts[:publish_to]` eu não gosto pq se `opts` não implementar o behaviour `[Access](https://hexdocs.pm/elixir/Access.html)` (não for map ou algo similar) a gente lança uma exception mais estranha que o `||` não trataria

Reply:
entendi. vendo a mensagem do erro atual não está tão ruim mesmo